### DISCUSSION TITLE
trigger work with superbase admin but not with client application using API call

### DISCUSSION THREAD

**Discussion by nisarg2023** [supabase/supabase#16607] [Questions]
this is my function 
**
BEGIN
    UPDATE auth. users
    SET email = NEW.email
    WHERE id = OLD.user_id;
    
    RETURN NEW;
END;
**

it calls when I update my email in public. users and update in auth. users it works properly with the superbase dashboard but does not work with the client application it gives error permission denied for table users. even it does not work with service_rolesecret

**Comment by GaryAustin1** [comment]
You would need to make the function "security definer" type so that it runs as the owner of the function (usually postgres role) versus the user calling it.  The REST API users have no access to auth schema.   "Security definer" is an option in the function UI, but can only be set when you create it.  You can update your current function in SQL though.

**Reply by nisarg2023** [reply]
can you give me more explanation with examples

**Reply by GaryAustin1** [reply]
https://supabase.com/docs/guides/database/functions#security-definer-vs-invoker
Also if you look at the user management starter code in the Dashboard SQL editor Welcome section you will see that trigger function is declared security definer type.  I assume you used something like that to populate your public.users table to begin with.

If you use the dashboard UI for functions, this is where the setting is at:
![image](https://github.com/supabase/supabase/assets/54564956/b4dbcd7b-4afc-49d5-b3c8-be3ff9b66f61)
But only when creating a function.
